airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py

```diff
@@ -0,0 +1,351 @@
+
+    primary_key = None
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "germany/age-groups"
+
+
+# Basic incremental stream
+class IncrementalRkiCovidStream(RkiCovidStream, ABC):
+
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> str:
+        """
+        TODO
+        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
+        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        return {}
+
+
+# source: germany/history/cases/:days | Incremental
+class GermanyHistoryCases(IncrementalRkiCovidStream):
+    """Docs: https://api.corona-zahlen.org/germany/germany/history/cases/:days"""
+
+    primary_key = None
+
+    def __init__(self, config, **kwargs):
+        super().__init__(**kwargs)
+        self.config = config
+        self._cursor_value = None
+
+    @property
+    def source_defined_cursor(self) -> bool:
+        return False
+
+    @property
+    def cursor_field(self) -> str:
+        return "date"
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return {self.cursor_field: str(self._cursor_value)}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = value[self.cursor_field]
+
+    def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(*args, **kwargs):
+            current_stream_state = record.get(self.cursor_field)
+            if self._cursor_value:
+                latest_state = record.get(self.cursor_field)
+                self._cursor_value = max(self._cursor_value, latest_state)
+            yield record
+            self._cursor_value = current_stream_state
+            assert self._cursor_value == current_stream_state
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        return response.json().get("data")
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        if self.config.get('cases_in_days'):
+            return "germany/history/cases/"+str(self.config.get('cases_in_days'))
```

Comment:
You're creating a dependency here for optional parameters and stream. This is not good. Maybe change the `cases_in_day` to `start_date` and use to all streams and create a function to convert date to integer days.

Reply:
added start_date and added a function that converts date to integer days.